lib/dep-types.ts

```diff
@@ -1,8 +1,11 @@
+import { AbbreviatedVersion } from "package-json";
+import { DepType, HasDependencySpecs } from "./types";
+
 // Dependency types.
 // We don't call out all of them, only the ones relevant to our behavior.
 // extraneous means not found in package.json files, prod means not dev ATM
-var depTypes = module.exports = function (depName, pkg) {
-  var type = null;
+function depTypes (depName: string, pkg: HasDependencySpecs) {
```

Comment:
would be great to type the return of this?

Reply:
I cannot export the type from this module yet (because of `export =` thing), and VS Code + TS are smart enough to infer this type for now. Added a TODO.